sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/context/propagation/OtTracePropagator.scala

```diff
@@ -0,0 +1,150 @@
+import org.typelevel.otel4s.trace.SpanContext
+import org.typelevel.otel4s.trace.SpanContext.SpanId
+import org.typelevel.otel4s.trace.SpanContext.TraceId
+import org.typelevel.otel4s.trace.TraceFlags
+import org.typelevel.otel4s.trace.TraceState
+import scodec.bits.ByteVector
+
+import java.util.Locale
+
+private final class OtTracePropagator extends TextMapPropagator[Context] {
+
+  import OtTracePropagator.Headers
+  import OtTracePropagator.Const
+
+  override def fields: List[String] = List(
+    Headers.TraceId,
+    Headers.SpanId,
+    Headers.Sampled
+  )
+
+  override def extract[A: TextMapGetter](ctx: Context, carrier: A): Context = {
+    val spanContext = for {
+      traceIdHex <- TextMapGetter[A].get(carrier, Headers.TraceId)
+      traceId <- ByteVector.fromHex(traceIdHex) if isValidTraceId(traceId)
+
+      spanHexId <- TextMapGetter[A].get(carrier, Headers.SpanId)
+      spanId <- ByteVector.fromHex(spanHexId) if isValidSpanId(spanId)
+
+      sampled <- TextMapGetter[A]
+        .get(carrier, Headers.Sampled)
+        .flatMap(_.toBooleanOption)
+
+    } yield SpanContext(
+      traceId = traceId,
+      spanId = spanId,
+      traceFlags = if (sampled) TraceFlags.Sampled else TraceFlags.Default,
+      traceState = TraceState.empty,
+      remote = true
+    )
+    // Baggage is only extracted if there is a valid SpanContext
+    spanContext match {
+      case Some(spanContext) =>
+        ctx
+          .updated(SdkContextKeys.SpanContextKey, spanContext)
+          .updated(SdkContextKeys.BaggageKey, decodeBaggage(carrier))
+      case None => ctx
+    }
+  }
+
+  override def inject[A: TextMapUpdater](ctx: Context, carrier: A): A = {
+
+    ctx.get(SdkContextKeys.SpanContextKey).filter(_.isValid) match {
+      case Some(spanContext) =>
+        val sampled =
+          if (spanContext.isSampled) Const.IsSampled else Const.NotSampled
+
+        val headers = Map(
+          Headers.TraceId -> spanContext.traceIdHex,
+          Headers.SpanId -> spanContext.spanIdHex,
+          Headers.Sampled -> sampled
+        ) ++ encodeBaggage(
+          ctx
+        ) // Baggage is only injected if there is a valid SpanContext
+
+        headers.foldLeft(carrier) { case (carrier, (key, value)) =>
+          TextMapUpdater[A].updated(carrier, key, value)
+        }
+
+      case None => carrier
+    }
+
+  }
+
+  override def toString: String = "OtTracePropagator"
+
+  private def encodeBaggage(ctx: Context) =
+    ctx
+      .get(SdkContextKeys.BaggageKey)
+      .map(_.asMap.map { case (key, entry) =>
+        Const.BaggagePrefix + key -> entry.value
+      })
+      .getOrElse(Map.empty)
+
+  private def decodeBaggage[A: TextMapGetter](carrier: A) = {
+    val keys = TextMapGetter[A].keys(carrier)
+    val prefixLength = Const.BaggagePrefix.length
+
+    keys.foldLeft(Baggage.empty) {
+      case (baggage, key)
+          if key.toLowerCase(Locale.ROOT).startsWith(Const.BaggagePrefix) =>
```

Comment:
` java.util.Locale` isn't available on the Scala.js platform. Currently, we take some risks by relying on the default Locale:
```suggestion
          if key.toLowerCase.startsWith(Const.BaggagePrefix) =>
```

This will fix the compilation issue

Reply:
Done